server/src/main/java/org/opensearch/action/bulk/BulkRequest.java

```diff
@@ -96,7 +96,7 @@ public class BulkRequest extends ActionRequest implements CompositeIndicesReques
     private String globalRouting;
     private String globalIndex;
     private Boolean globalRequireAlias;
-    private int batchSize = 1;
+    private int batchSize = Integer.MAX_VALUE;
```

Comment:
@andrross  @dblock I believe as we discussed, this parameter has to be deprecated (jn `2.x `) and removed from `3.x` since it has nothing to do with bulk but ingestion processing.

Reply:
Agree this should be deprecated in 2.x and removed in 3.0

Reply:
@chishui could you please work towards this goal? thank you

Reply:
Oh, I planned to submit a separate PR for deprecation, I can have it in this PR as well

Reply:
BTW, could you share some guidances on deprecating a parameter, my understanding is:
- First PR, need to backport to 2.x:
  - Reporting a deprecation log whenever the parameter is used
- Second PR, need to backport to 3.0
  - Remove the parameter and backport to 3.0 
- Documentation PR
  - Mark the parameter as deprecated in documentation.

Reply:
Thanks @chishui , basically there are two ways, the one you've described or:
- remove the parameter in 3.0, but ...
-  ... backport as deprecation in 2.x (and mark the parameter as deprecated in documentation)